controllers/quay/quayregistry_controller.go

```diff
@@ -120,10 +120,91 @@ func (r *QuayRegistryReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, nil
 	}
 
-	created := v1.GetCondition(quay.Status.Conditions, v1.ConditionComponentsCreated)
-	if created != nil && created.Reason == v1.ConditionReasonMigrationsInProgress {
-		log.Info("migrations in progress, skipping reconcile")
-		return ctrl.Result{}, nil
+	created := v1.GetCondition(updatedQuay.Status.Conditions, v1.ConditionComponentsCreated)
+	migrationInProgress := created != nil && created.Reason == v1.ConditionReasonMigrationsInProgress
+	if migrationInProgress {
+		log.Info("++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++")
+		log.Info("migration currently in progress!")
+		log.Info(fmt.Sprintf("currentVersion is: '%s'", updatedQuay.Status.CurrentVersion))
+
+		log.Info("checking Quay upgrade `Job` completion")
+
+		var upgradeJob batchv1.Job
+		err := r.Client.Get(
+			ctx,
+			types.NamespacedName{
+				Name:      updatedQuay.GetName() + "-" + v1.QuayUpgradeJobName,
+				Namespace: updatedQuay.GetNamespace(),
+			},
+			&upgradeJob,
+		)
+		if err != nil {
+			log.Error(err, "could not retrieve Quay upgrade `Job`")
+			return ctrl.Result{RequeueAfter: time.Second * 1}, nil
+		}
+
+		if upgradeJob.Status.Active == 1 {
+			return ctrl.Result{RequeueAfter: time.Second * 1}, nil
+		}
+
+		if upgradeJob.Status.Succeeded == 1 {
+			log.Info("Quay upgrade complete, updating `status.currentVersion`")
+
+			var freshQuay v1.QuayRegistry
+			if err := r.Client.Get(ctx, req.NamespacedName, &freshQuay); err != nil {
+				log.Error(err, "could not retrieve QuayRegistry")
+				return ctrl.Result{RequeueAfter: time.Second * 1}, nil
+			}
+			qcopy := freshQuay.DeepCopy()
+
+			msg := "All registry components created"
+			condition := v1.Condition{
+				Type:               v1.ConditionComponentsCreated,
+				Status:             metav1.ConditionTrue,
+				Reason:             v1.ConditionReasonComponentsCreationSuccess,
+				Message:            msg,
+				LastUpdateTime:     metav1.Now(),
+				LastTransitionTime: metav1.Now(),
+			}
+			qcopy.Status.Conditions = v1.SetCondition(qcopy.Status.Conditions, condition)
+			qcopy.Status.CurrentVersion = v1.QuayVersionCurrent
+			r.EventRecorder.Event(qcopy, corev1.EventTypeNormal, string(v1.ConditionReasonHealthChecksPassing), msg)
```

Comment:
I think it is safe to remove this (if tests are depending on this then they should be fixed). What do you think?

Reply:
Yup, let's remove it and see where we end up.